src/cdk-lib/bedrock/agent.ts

```diff
@@ -80,6 +85,109 @@ export interface GuardrailConfiguration {
   readonly guardrailId?: string;
 }
 
+export interface IAgent extends IResource {
+
+  /**
+   * The name of the agent.
+   */
+  readonly name: string;
+
+  /**
+   * The IAM role for the agent.
+   */
+  readonly role: iam.Role;
+
+  /**
+   * The unique identifier of the agent.
+   */
+  readonly agentId: string;
+
+  /**
+   * The ARN of the agent.
+   */
+  readonly agentArn: string;
+
+  /**
+   * Instance of Agent
+   */
+  readonly agentInstance: bedrock.CfnAgent;
+
+  /**
+   * The unique identifier of the agent alias.
+   */
+  readonly aliasId?: string;
+
+  /**
+   * The ARN of the agent alias.
+   */
+  readonly aliasArn?: string;
+
+  /**
+   * The name for the agent alias.
+   */
+  readonly aliasName?: string;
+
+  /**
+   * The version for the agent.
+   */
+  readonly agentVersion: string;
+
+  /**
+   * A list of action groups associated with the agent.
+   */
+  actionGroups: bedrock.CfnAgent.AgentActionGroupProperty[];
+
+  /**
+   * A list of KnowledgeBases associated with the agent.
+   */
+  knowledgeBases: bedrock.CfnAgent.AgentKnowledgeBaseProperty[];
```

Comment:
I think it would be better to expose this as `IKnowledgeBases[]`. ([link to best practices](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#imports:~:text=Do%20not%20%E2%80%9Cleak%E2%80%9D%20the%20details%20or%20types%20of%20the%20CFN%20layer%20when%20defining%20your%20construct%20API.%20In%20almost%20all%20cases%2C%20a%20richer%20object%2Doriented%20API%20can%20be%20exposed%20to%20encapsulate%20the%20low%2Dlevel%20surface%20%5Bawslint%3Aprops%2Dno%2Dcfn%2Dtypes%5D.))

Reply:
Not sure I am following this one. Feel free to drop a line of code of how you think it should look like

Reply:
An interface like `IConstructName` is basically the structure other constructs can leverage. The best practices tell us that we should expose higher level constructs and avoid exposing low-level CloudFormation types like `bedrock.CfnAgent.AgentKnowledgeBaseProperty`. So in the interface we should use something like.

Reply:
Refactored